### PR TITLE
Add clearing the text in the search bar on the leaving the search page.

### LINKED ISSUE #232: Reset search bar
Using the side nav after searching should clear the search bar input

### PR DESCRIPTION
**Pull request by marekdano**
I found the way how to clear the term in the search bar on the leaving the search bar. I call the resetSearch method to set the searchTerm to empty string. The call is in the componentWillUnmount method in the pages/search component.
closes #232

### CODE REVIEW

File: src/pages/search.jsx [hunk 1]
```diff
@@ -16,6 +16,7 @@ import {
 const propTypes = {
   isSearching: PropTypes.bool,
   lastPage: PropTypes.string,
+  resetSearch: PropTypes.func,
```
Comment (Bouncey) [inline src/pages/search.jsx]: `PropTypes.func.isRequired`

If the component try's to unMount and this function is not available, the app will crash

File: src/pages/search.jsx [hunk 2]
```diff
@@ -42,27 +43,38 @@ function shouldShowResults(conditions) {
...
-  ) {
-    return <NoSupport />;
+class SearchPage extends PureComponent {
+  constructor() {
+    super();
+  }
+
+  componentWillUnmount() {
+    const { resetSearch } = this.props;
+    resetSearch('');
```
Comment (Bouncey) [inline src/pages/search.jsx]: This action expects no arguments, you are creating an empty string for no reason

File: src/pages/search.jsx [hunk 3]
```diff
@@ -42,27 +43,38 @@ function shouldShowResults(conditions) {
...
+    return (
+      <div>
+        <Helmet>
+          <title>Search | freeCodeCamp Guide</title>
+        </Helmet>
+        <h2 className='colourDarkGrey'>Search Results</h2>
+        {
+          (isSearching && !results.length) ?
+            <ResultsSkeleton /> :
+            shouldShowResults({ results, lastPage, searchTerm })
```
Comment (Bouncey) [inline src/pages/search.jsx]: Now this page is a `class`, we should make this function a method on the class. By doing so, we will not need to pass props to it manually, as they will be available via `this.props`

File: src/pages/search.jsx (hunk 1 above)
Comment (marekdano) [reply src/pages/search.jsx]: Ok, I modified it.

File: src/pages/search.jsx (hunk 2 above)
Comment (marekdano) [reply src/pages/search.jsx]: Good, I'll remove the empty string.

File: src/pages/search.jsx (hunk 3 above)
Comment (marekdano) [reply src/pages/search.jsx]: Ok, I understand your comment. Just one question. What is the best practice? Moving the method "shouldShowResults" before "componentWillUnmount" or after.

Comment (Bouncey) [reply src/pages/search.jsx]: I think we should go with 
```jsx
class Search extends PureComponent {
  // constructor

  // component lifestyle hooks

  // component methods

  // render() {...}
}
```

Thanks for asking @marekdano

Comment (marekdano) [reply src/pages/search.jsx]: @Bouncey Thanks for your answer. I've just pushed my code. Hopefully it's good ;)

**Comment by marekdano** [comment]
Let me know if you think that the code should be improved.

**Comment by Bouncey** [comment]
Great Job @marekdano 

Sorry for the delay and thank you for your patience!

**Pull request merged**